multicluster/controllers/multicluster/core/resourceimport_controller.go

```diff
@@ -0,0 +1,408 @@
+	// TODO: Must check whether this ResourceImport must be reconciled by this member cluster. Check `spec.clusters` field.
+	if r.localClusterClient == nil {
+		return ctrl.Result{}, errors.New("localClusterClient has not been initialized properly, no local cluster client")
+	}
+
+	if r.remoteCommonArea == nil {
+		return ctrl.Result{}, errors.New("remoteCommonArea has not been initialized properly, no remote common area")
+	}
+
+	var resImp multiclusterv1alpha1.ResourceImport
+	err := r.remoteCommonArea.Get(ctx, req.NamespacedName, &resImp)
+	var isDeleted bool
+	if err != nil {
+		isDeleted = apierrors.IsNotFound(err)
+		if !isDeleted {
+			klog.InfoS("unable to fetch ResourceImport", "resourceimport", req.NamespacedName.String(), "err", err)
+			return ctrl.Result{}, err
+		} else {
+			resImpObj, exist, err := r.installedResImports.GetByKey(req.NamespacedName.String())
+			if exist {
+				resImp = resImpObj.(multiclusterv1alpha1.ResourceImport)
+			} else {
+				// service_controller will reconcile and clean up MCS Service/ServiceImport, so it's ok to return nil here
+				klog.ErrorS(err, "no cached data for ResourceImport", "resourceimport", req.NamespacedName.String())
+				return ctrl.Result{}, nil
+			}
+		}
+	}
+
+	switch resImp.Spec.Kind {
+	case common.ServiceImportKind:
+		if isDeleted {
+			return r.handleResImpDeleteForService(ctx, &resImp)
+		}
+		return r.handleResImpUpdateForService(ctx, &resImp)
+	case common.EndpointsKind:
+		if isDeleted {
+			return r.handleResImpDeleteForEndpoints(ctx, &resImp)
+		}
+		return r.handleResImpUpdateForEndpoints(ctx, &resImp)
+	}
+	// TODO: handle for other ResImport Kinds
+	return ctrl.Result{}, nil
+}
+
+func (r *ResourceImportReconciler) handleResImpUpdateForService(ctx context.Context, resImp *multiclusterv1alpha1.ResourceImport) (ctrl.Result, error) {
+	svcImpName := types.NamespacedName{Namespace: resImp.Spec.Namespace, Name: resImp.Spec.Name}
+	svcName := types.NamespacedName{Namespace: resImp.Spec.Namespace, Name: common.AntreaMCSPrefix + resImp.Spec.Name}
+	klog.InfoS("Updating Service and ServiceImport corresponding to ResourceImport",
+		"service", svcName.String(), "serviceimport", svcImpName.String(), "resourceimport", klog.KObj(resImp))
+
+	svc := &corev1.Service{}
+	err := r.localClusterClient.Get(ctx, svcName, svc)
+	svcNotFound := apierrors.IsNotFound(err)
+	if err != nil && !svcNotFound {
+		return ctrl.Result{}, err
+	}
+	if !svcNotFound {
+		if _, ok := svc.Annotations[common.AntreaMCServiceAnnotation]; !ok {
+			err := errors.New("unable to import Service which conflicts with existing one")
+			klog.ErrorS(err, "", "service", klog.KObj(svc))
+			return ctrl.Result{}, err
+		}
+	}
+	mcsObj := getMCService(resImp)
+	if svcNotFound {
+		err := r.localClusterClient.Create(ctx, mcsObj, &client.CreateOptions{})
+		if err != nil {
+			klog.ErrorS(err, "fail to import Service", "service", klog.KObj(mcsObj))
```

Comment:
failed to create Service?

Reply:
service -> Service

Reply:
for the second comments, I suppose the key for `klog.ErrorS` should be lower case?

Reply:
done